### ISSUE TITLE
[BUG] Recording dates are out of sync and change on their own

### ISSUE THREAD

**Issue by Arkanattahari** [Ivy-Apps/ivy-wallet#3186]
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

When I record expenses on today's date May 7, why is it recorded on UI as May 6. And when I fix it by editing back to May 7, it brings my record to May 5. And on the edit page, it shows that it was recorded on the 6th.

I've a screen record, if needed.

### To Reproduce

1. Record Expenses on May 7
2. The date appeared instead became May 6
3. I edited it back to the 7th
4. My recording was instead thrown to May 5

### Screenshots

![InShot_20240507_063646074](https://github.com/Ivy-Apps/ivy-wallet/assets/169105946/ee50fa88-9049-4c27-a5f7-946b38ce3471)
![InShot_20240507_063706861](https://github.com/Ivy-Apps/ivy-wallet/assets/169105946/96ef317d-7473-492f-b0b0-e072f7633cb6)
![InShot_20240507_063751048](https://github.com/Ivy-Apps/ivy-wallet/assets/169105946/80d664c5-74d0-465c-bc1f-4cb98888ed9b)

### App version

4.6.3 (163) from Google Play

### Smartphone

Xiaomi Redmi Note 7, Android 13

**Comment by ivywallet** [comment]
Thank you @Arkanattahari for raising Issue #3186! 🚀
What's next? Read our **[Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) 📚**.

_Tagging @ILIYANGERMANOV for review & approval 👀_

**Comment by ILIYANGERMANOV** [comment]
Yeah, there are like 10-20 reports of the same issue. The core problem is that timezones are messed up and conversions UTC <> local-time are incorrect
